### ISSUE TITLE
Utilize the taxonomy data of the tech_catalog to create a link to IT standards (software)

### ISSUE THREAD

**Issue by hatfieldjm4** [GSA/GEAR3#229]
Use case; if I want to see which IT standards that are already approved for use for satisfying a project management capability, I'd like to be able to search a software taxonomy for "project management" to return any software titles that are approved for use so that I may request it. 

We already have a tp_Taxonomy table in the tech_catalog schema that has a reference to the tp_softwareProduct table that we could use (and the references are already established for us). We would just need to check out the robustness of the taxonomy to determine if it will be useful. 

This would be very similar to how business capabilities tie to business systems. The technology taxonomy should be its own report that you search a software function, like "Portfolio Management" and it returns the subcategory of "Project and Portfolio Management". Then, the user can click on this and see any IT standards that align to that taxonomy to determine whether they are approved.

**Comment by hatfieldjm4** [comment]
@joshsjurik Hi Josh, at our meeting today, I can walk you through the parent/child relationships within this taxonomy and how we want to go about displaying it in the IT Standards report/modal views.

**Comment by hatfieldjm4** [comment]
@joshsjurik In the dev environment, you'll notice that I added a data field on obj_technology to house the Taxonomy Id that links back to the tech_catalog.tp_Taxonomy table. I'm debating whether we actually need this data field, though, as we can simply derive this information with a JOIN that we can use when a user adds or edits an IT standard's "Software Product" data: 
```SELECT 
gear_schema.obj_technology.Id as TechId,
gear_schema.obj_technology.Keyname as TechName,
gear_schema.obj_technology.softwareProduct as TechSoftwareProduct,
tech_catalog.tp_Taxonomy.id as TaxonomySubcategoryId,
tech_catalog.tp_Taxonomy.subcategory as Subcategory,
tech_catalog.tp_Taxonomy.category as Category,
tech_catalog.tp_Taxonomy.categoryGroup as CategoryGroup,
tech_catalog.tp_Taxonomy.categoryId as CategoryId,
tech_catalog.tp_Taxonomy.description as TaxonomyDescription, 
tech_catalog.tp_Taxonomy.softwareOrHardware as SoftwareOrHardware,
tech_catalog.tp_SoftwareProduct.id as SoftwareProductId,
tech_catalog.tp_SoftwareProduct.taxonomy as SoftwareProductTaxonomy
FROM tech_catalog.tp_SoftwareProduct
INNER JOIN tech_catalog.tp_Taxonomy on tech_catalog.tp_SoftwareProduct.taxonomy = tech_catalog.tp_Taxonomy.id
INNER JOIN gear_schema.obj_technology on tech_catalog.tp_SoftwareProduct.id  = gear_schema.obj_technology.softwareProduct```

What are your thoughts?

**Comment by joshsjurik** [comment]
@hatfieldjm4  I don't think it really matters since we'd be doing a join either way. We're either joining the tech table onto the taxonomy table or the taxonomy table onto the tech table to display the data we'd need. I think using the join you have above is perfectly fine.

**Comment by hatfieldjm4** [comment]
For awareness, the grandparent, parent, child relationship (in that order) is: 
- Category group
- Category
- Subcategory

The category group and subcategory data fields are never null, but the category data field can be. For example, you'll notice a "categoryId" field. If you query the Taxonomy table on this id, it provides a separate object defining what is meant by that parent-level "Category", however, it lists the Category name in the "Subcategory" field (a bit confusing) and lists a blank for the "Category" field when you look up the categoryId.
